packages/@react-aria/test-utils/src/combobox.ts

```diff
@@ -0,0 +1,191 @@
+      if (triggerBehavior !== 'focus') {
+        await this.user.keyboard('{ArrowDown}');
+      }
+    } else if (this._interactionType === 'touch') {
+      if (triggerBehavior === 'focus') {
+        await this.user.pointer({target: this.getCombobox(), keys: '[TouchA]'});
+      } else {
+        await this.user.pointer({target: this.getTrigger(), keys: '[TouchA]'});
+      }
+    }
+
+    await waitFor(() => {
+      if (!isDisabled && this.getCombobox().getAttribute('aria-controls') == null) {
+        throw new Error('No aria-controls found on combobox trigger element.');
+      } else {
+        return true;
+      }
+    });
+    let listBoxId = this.getCombobox().getAttribute('aria-controls');
+    await waitFor(() => {
+      if (!isDisabled && (!listBoxId || document.getElementById(listBoxId) == null)) {
+        throw new Error(`Listbox with id of ${listBoxId} not found in document.`);
+      } else {
+        return true;
+      }
+    });
+  };
+
+  selectOption = async (opts: {option?: HTMLElement, optionText?: string, triggerBehavior?: 'focus' | 'manual'} = {}) => {
+    let {optionText, option, triggerBehavior} = opts;
+    if (!this.getCombobox().getAttribute('aria-controls')) {
+      await this.open({triggerBehavior});
+    }
+
+    let listbox = this.getListbox();
+    if (listbox) {
+      if (!option && optionText) {
+        option = within(listbox).getByText(optionText);
+      }
+
+      // TODO: keyboard method of selecting the the option is a bit tricky unless I simply simulate the user pressing the down arrow
+      // the required amount of times to reach the option. For now just click the option even in keyboard mode
+      if (this._interactionType === 'mouse' || this._interactionType === 'keyboard') {
+        await this.user.click(option);
+      } else {
+        await this.user.pointer({target: option, keys: '[TouchA]'});
+      }
+
+      if (option && option.getAttribute('href') == null) {
+        await waitFor(() => {
+          if (document.contains(listbox)) {
+            throw new Error('Expected listbox element to not be in the document after selecting an option');
+          } else {
+            return true;
+          }
+        });
+      }
+    } else {
+      throw new Error("Attempted to select a option in the combobox, but the listbox wasn't found.");
+    }
+  };
+
+  close = async () => {
+    let listbox = this.getListbox();
+    if (listbox) {
+      act(() => this.getCombobox().focus());
+      await this.user.keyboard('[Escape]');
+
+      await waitFor(() => {
+        if (document.contains(listbox)) {
+          throw new Error('Expected listbox element to not be in the document after selecting an option');
+        } else {
+          return true;
+        }
+      });
+    }
+  };
+
+  getCombobox = () => {
+    if (!this._combobox) {
+      throw new Error('Combobox input element hasn\'t been set yet. Did you call `setElement()` yet?');
+    }
+
```

Comment:
add a check for `this._combobox.isConnected`?

Reply:
hm, I'm going back and forth on this. What if the user want to explicitly check that the combobox doesn't exist in their test? I don't want to throw an error in that case

Reply:
had a thought on this one, it's possible that the user accidentally completely unmounts and replaces their component with a new instance. so I think checking for isConnected could be useful to track that

side note, maybe this is the reason that PO's in selenium always retrieve the element new for every interaction, maybe we should do the same? https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/

Reply:
In that case, the root element itself has changed so we can't query for it properly in the util right?

Reply:
depends on the root that was given
could've been application root

just because an element is unmounted does not mean it's gc'd, we still have a reference to the unattached node, which is problematic

Reply:
My running assumption with the root that the user provides to the tester util is that it is either the target element itself (aka the role=combobox) or the immediate wrapper (aka what a ref provided to <Combobox /> would return). I would hope that the user wouldn't provide the actual application root just because it could be a bit hairy if there were multiple comboboxes in their app or what not.